src/main/java/org/folio/dataexp/service/export/strategies/HoldingsExportAllStrategy.java

```diff
@@ -23,112 +21,131 @@
+      List<MarcRecordEntity> marcRecords) {
+    var externalIds = marcRecords.stream().map(MarcRecordEntity::getExternalId).collect(Collectors.toSet());
+    createMarc(externalIds, exportStatistic, mappingProfile, exportFilesEntity.getJobExecutionId(), new HashSet<>(),
+        marcRecords);
+  }
+
+  private void processFolioHoldings(JobExecutionExportFilesEntity exportFilesEntity, ExportStrategyStatistic exportStatistic, MappingProfile mappingProfile,
+      List<HoldingsRecordEntity> folioHoldings) {
+    var result = getGeneratedMarc(folioHoldings, mappingProfile, exportFilesEntity.getJobExecutionId());
+    saveMarc(result, exportStatistic);
+  }
+
+  private void processFolioHoldingsDeleted(JobExecutionExportFilesEntity exportFilesEntity, ExportStrategyStatistic exportStatistic,
+      MappingProfile mappingProfile, List<HoldingsRecordEntity> folioHoldings) {
+    var result = getGeneratedMarc(folioHoldings, mappingProfile, exportFilesEntity.getJobExecutionId());
+    saveMarc(result, exportStatistic);
   }
 
-  private Slice<HoldingsRecordEntity> chooseSlice(JobExecutionExportFilesEntity exportFilesEntity, ExportRequest exportRequest, Pageable pageble) {
+  private Slice<HoldingsRecordEntity> nextFolioSlice(JobExecutionExportFilesEntity exportFilesEntity, ExportRequest exportRequest, Pageable pageble) {
     if (Boolean.TRUE.equals(exportRequest.getSuppressedFromDiscovery())) {
-      return holdingsRecordEntityRepository.findByIdGreaterThanEqualAndIdLessThanEqualOrderByIdAsc(exportFilesEntity.getFromId(),
-        exportFilesEntity.getToId(), pageble);
+      return folioHoldingsAllRepository.findFolioHoldingsAllNonDeleted(exportFilesEntity.getFromId(), exportFilesEntity.getToId(),
+          pageble);
     }
-    return holdingsRecordEntityRepository.findAllWhenSkipDiscoverySuppressed(exportFilesEntity.getFromId(),
-      exportFilesEntity.getToId(), pageble);
+    return folioHoldingsAllRepository.findFolioHoldingsAllNonDeletedNonSuppressed(exportFilesEntity.getFromId(),
+        exportFilesEntity.getToId(), pageble);
   }
 
-  private List<HoldingsRecordEntity> holdingsDeletedToHoldingsEntities(List<HoldingsRecordDeletedEntity> holdingsDeleted) {
-    return holdingsDeleted.stream()
-      .map(hold -> new HoldingsRecordEntity().withId(UUID.fromString(getAsJsonObject(getAsJsonObject(hold.getJsonb()).get()
-          .getAsString(DELETED_AUDIT_RECORD)).get()
-          .getAsString("id")))
-        .withJsonb(getAsJsonObject(hold.getJsonb()).get()
-          .getAsString(DELETED_AUDIT_RECORD))
-        .withInstanceId(UUID.fromString(getAsJsonObject(getAsJsonObject(hold.getJsonb()).get()
-          .getAsString(DELETED_AUDIT_RECORD)).get()
-          .getAsString("instanceId"))))
-      .toList();
+  private Slice<MarcRecordEntity> nextMarcSlice(JobExecutionExportFilesEntity exportFilesEntity, ExportRequest exportRequest, Pageable pageble) {
+    if (Boolean.TRUE.equals(exportRequest.getSuppressedFromDiscovery())) {
```

Comment:
@obozhko-folio 
I believe the boolean values equality check is not needed.
(exportRequest.getSuppressedFromDiscovery()) - is enough for the condition check.

Reply:
exportRequest.getSuppressedFromDiscovery() can be null and it leads to NPE